src/status_im/ui/screens/wallet/request/events.cljs

```diff
@@ -16,12 +16,13 @@
   (fn [{{:wallet/keys [request-transaction]} :db} [_ {:keys [whisper-identity]}]]
     {:dispatch-n [[:navigate-back]
                   [:navigate-to-clean :chat-list]
-                  [:add-chat-loaded-event whisper-identity [::wallet-send-chat-request (:amount request-transaction)]]
+                  [:add-chat-loaded-event whisper-identity [::wallet-send-chat-request (str (:amount request-transaction))]]
                   [:start-chat whisper-identity]]}))
 
 (handlers/register-handler-fx
-  :wallet-validate-request-amount
-  (fn [{{:wallet/keys [request-transaction] :as db} :db} _]
-    (let [amount (:amount request-transaction)
-          error (wallet.db/get-amount-validation-error amount)]
-      {:db (assoc-in db [:wallet/request-transaction :amount-error] error)})))
+  :wallet.request/set-and-validate-amount
+  (fn [{:keys [db]} [_ amount]]
+    (let [{:keys [value error]} (wallet.db/parse-amount amount)]
+      {:db (-> db
+               (assoc-in [:wallet/request-transaction :amount] (money/ether->wei value))
+               (assoc-in [:wallet/request-transaction :amount-error] error))})))
```

Comment:
also you could update :wallet/request-transaction and assoc two fields, but don't insist

Reply:
I didn't want to make too many unrelated changes, already too much